sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -880,6 +909,11 @@ size_t CurlSession::OnRead(uint8_t* buffer, size_t count, Context const& context
     return 0;
   }
 
+  // If we no longer have a connection, read 0 bytes.
+  if (!m_connection)
+  {
+    return 0;
+  }
```

Comment:
This is needed because if `m_connection` is null, then this code will AV:

```cpp
totalRead = m_connection->ReadFromSocket(....);
```

I normally prefer deterministic behaviour over access violations.